app_feup/lib/view/Pages/HomePageView.dart

```diff
@@ -19,15 +17,20 @@ class HomePageView extends GeneralPageView {
             style: Theme.of(context).textTheme.title,
           ),
         ),
-        new GenericCard(
-            title: "Exames",
-            child: new ExamCard()
+        GestureDetector(
+          onTap: () {
+            Navigator.pushReplacementNamed(context, '/Mapa de Exames');
+          },
+          child: ExamCard(),
         ),
-        new GenericCard(
-          title: "Horário",
-          child: new ScheduleCard(),
-        ),
-        //Cards go here
+
+        GestureDetector(
+          onTap: () {
+            Navigator.pushReplacementNamed(context, '/Horário');
+          },
+          child: ScheduleCard(),
```

Comment:
What are the pro's and con's of having the gesture detector be a part of the card widgets (ExamCard and ScheduleCard)

Reply:
The pros would be having a better-organized code as you said. 
The only disadvantage I can see it's in the case where we do not want to be redirected to the corresponding page. However, this case shouldn't happen in our current plans for the app.